mlflow/utils/conda.py

```diff
@@ -133,10 +181,29 @@ def get_or_create_conda_env(conda_env_path, env_id=None, capture_output=False):
             )
         )
 
-    env_names = _list_conda_environments()
+    conda_extra_envs = _get_conda_extra_envs(env_root_dir)
+
     project_env_name = _get_conda_env_name(conda_env_path, env_id)
-    if project_env_name not in env_names:
-        _logger.info("=== Creating conda environment %s ===", project_env_name)
+    if env_root_dir is not None:
+        # Append a suffix "-isolated" because if a conda env name exist in
+        # default conda env root dir, then if we set new "CONDA_ENVS_PATH" and run "conda env create"
+        # with the same conda env name, "CondaValueError: prefix already exists" error will
+        # be raised.
```

Comment:
Are you sure this happens? Is there a way to exclude the default conda env root dir? The docs (https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-environment-directories-envs-dirs) seem to imply that the default location should no longer be checked if `CONDA_ENVS_PATH` is specified:

```
Specify environment directories (envs_dirs)[](https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-environment-directories-envs-dirs)
Specify directories in which environments are located. If this key is set, the root prefix envs_dir is not used unless explicitly included. This key also determines where the package caches are located.
```